crates/maybenot/src/framework.rs

```diff
@@ -87,6 +85,10 @@ where
     blocking_duration: T::Duration,
     blocking_started: T,
     blocking_active: bool,
+    // internal signaling: if None then we don't signal, Some(None) means all
+    // machines receive the signal, Some(Some(...)) means one machine doesn't
+    // (the one that sent the signal)
+    signal_pending: Option<Option<usize>>,
```

Comment:
It's great that you have documented what each value means! However, the comment won't be immediately available where this value is "used", so can be hard to understand there.

The current implementation works! But I would maybe consider replacing the inner `Option<usize>` with a custom type something like:

```rust
enum SignalTarget {
    All,
    AllExcept(usize),
}
```

Reply:
This is a good idea! Done.